server/submissionWorkflow/queries.ts

```diff
@@ -0,0 +1,42 @@
+import { Collection, SubmissionWorkflow } from 'server/models';
+// import { SubmissionWorkflow as SubmissionWorkflowType } from 'types';
+
+export const createSubmissionWorkflow = async ({
+	collectionId,
+	enabled,
+	instructions,
+	afterSubmittedText,
+	email,
+	layoutBlock,
+}) => {
+	return Collection.findOne({ where: { id: collectionId } }).then(() =>
+		SubmissionWorkflow.create({
+			collectionId,
+			enabled,
+			instructions,
+			afterSubmittedText,
+			email,
+			layoutBlock,
+		}),
+	);
+};
+
+export const updateSubmissionWorkflow = async (inputValues, updatePermissions) => {
```

Comment:
5️⃣ There are a few things that need some attention here in `updateSubmissionWorkflow`:

As I mentioned over in `permissions.ts`, you can simplify this logic by noting that the `updatePermissions` is constant for all users that have _any_ permissions to call this function, and remove them entirely. Another benefit of this is that it decouples queries from any notion of permissions, so they can be easily called elsewhere on the server, outside of permissions-aware API routes.

Now we want a type for the lone `inputValues` argument. I'd rename this argument to `update`, and give it the type  `Partial<Omit<types.SubmissionWorkflow, 'id' | 'createdAt' | 'updatedAt'>>` — and probably factor the `Omit<...>` into a type that's shared with the create query above.

This puts some bounds on what TypeScript will allow us to pass in here, but that does nothing to prevent runtime code from sneaking extra values into the SQL query. To do that, let's unpack all the variables we actually want to allow folks to change:

```ts
const { collectionId, enabled, instructionsText, emailText, ... } = update;
await SubmissionWorkflow.update({ enabled, instructionsText, ... }, { where: { collectionId } });
```

It's good practice when writing Sequelize queries to pass in an object literal that explicitly lays out the props that are being updated. When we write `Model.update(objectFromElsewhere)` I'm always a little suspicious that `objectFromElsewhere` might have some nefarious, unanticipated props — even when it's defined very nearby.